internal/ingest/ingest.go

```diff
@@ -160,6 +163,7 @@ func NewIngester(cfg config.Ingest, h host.Host, idxr indexer.Interface, reg *re
 		providerAdChainStaging:  make(map[peer.ID]*atomic.Value),
 		toWorkers:               NewPriorityQueue(),
 		closeWorkers:            make(chan struct{}),
+		deletedProviders:        make(map[string]bool),
```

Comment:
This can be local to `ingestWorkerLogic`.  I think it is only necessary to remember deleted providers during the processing of the current chain.